### ISSUE TITLE
Type Error, wants int

### ISSUE THREAD

**Issue by darinlarimore** [verbb/gift-voucher#113]
### Describe the bug

I'm getting a TypeError when trying to duplicate a voucher code. 
<img width="1500" alt="Screenshot 2023-02-04 at 2 28 31 PM" src="https://user-images.githubusercontent.com/20300635/216786375-efb6e98b-cbe1-445b-8c3f-0a0e62dd90a1.png">

### Steps to reproduce

1. Create voucher code
2. Try to duplicate voucher using "Save as a new code" button.

<img width="265" alt="Screenshot 2023-02-04 at 2 23 40 PM" src="https://user-images.githubusercontent.com/20300635/216786451-17375763-4ad5-4835-ba2a-4984a1db8001.png">

### Craft CMS version

4.3.6.1

### Plugin version

3.0.6

### Multi-site?

No

**Comment by darinlarimore** [comment]
I have an attempt to fix this here! #114

**Comment by engram-design** [comment]
Fixed for the next release (thanks for the PR). To get the fix early, change your `verbb/gift-voucher` requirement in `composer.json` to:

```
"require": {
  "verbb/gift-voucher": "dev-craft-4 as 3.0.6",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [3.0.17](https://github.com/verbb/gift-voucher/releases/tag/3.0.7)
